_docs/extending-expired-ca-cert.md

```diff
@@ -0,0 +1,61 @@
+---
+layout: post
+title: Extending Expired Puppet CA Certificate
+date: 2025-08-06
+summary: How to extend the expiration date of the Puppet CA certificate
+---
+
+## Using a Bolt Module
+
+If you're using Bolt in your environment, there's a
+[ca_extend](https://github.com/puppetlabs/ca_extend) module designed for this
+task. Follow the module’s documentation to perform the extension in a supported
+and automated way.
+
+## Manual Process
+
+The [ca_extend repository](https://github.com/puppetlabs/ca_extend) also includes a standalone [extend.sh](https://github.com/puppetlabs/ca_extend/blob/main/files/extend.sh) script that can be run manually to generate a new CA certificate with an extended expiration date.
+
+The script performs the following steps:
+
+1. Sets up a temporary SSL environment.
+2. Generates a new CA certificate with the expiration set 15 years into the future.
+3. Writes the new certificate to the directory returned by:
+  ```
+  puppet config print --section master cacert
+  ```
+
+The new file will be named using the format: `ca_crt-expires-<NEW_END_DATE>.pem`. This allows you to distinguish it from the currently active certificate without overwriting anything by default.
+
+## Steps After Running the Script
+
+1. Examine the end date of the new certificate by executing:
+
+```
+openssl x509 -in <PATH_TO_NEW_KEY> -noout -text
+```
+
+Confirm the issuer matches your existing CA and that the expiration date is 15 years in the future.
+
+2. Back up the current certificate:
+
+```
+# Assuming the directory where the CA is stored is /etc/puppetlabs/puppet/ssl/ca
+cp /etc/puppetlabs/puppet/ssl/ca/ca_crt.pem /etc/puppetlabs/puppet/ssl/ca/ca_crt.pem.bak.$(date +%F)
+```
+3. Restart the puppet server.
+```
+systemctl restart puppetserver
+```
+
+# Puppet agents
+
+## Agents on Puppet 8 or newer
+Agents running Puppet 8+ will automatically fetch the updated CA certificate according to the [ca_refresh_interval](https://help.puppet.com/core/current/Content/PuppetCore/Markdown/configuration.htm#ca_refresh_interval) setting.
```

Comment:
I am not sure about what happen if the CA has expired. There is some Trust On First Use when you initially get a CA certificate, but if the local copy is expired will it ignore it?  That would be quite terrible IMHO…  Maybe worth testing.